AG-NONE Update config files to mutes TS compiler errors

olegat opened the pull request.

olegat committed:
AG-NONE Update config files to mutes TS compiler errors

olegat committed:
AG-NONE Ignore test directories

olegat committed:
AG-NONE Remove trailing comma

alantreadway commented:
Just to note - I'd prefer we went the other way, enforced type checking on tests and fix up the tests.

olegat commented:
> Just to note - I'd prefer we went the other way, enforced type checking on tests and fix up the tests.

I agree. There are a lot of errors in the tests right now. I'll have a look at that

olegat committed:
AG-NONE Add test/utils.ts type ChartOrProxy

We have several places where the `chart` variable is of type
AgChartInstance, so create a type union for all test utils.

olegat committed:
AG-NONE CR: Enable TS compiler on tests

olegat committed:
AG-NONE Fix many TS errors in test files

olegat commented on libraries/ag-charts-test/src/canvas/mock-canvas.ts:
```diff
@@ -23,6 +27,7 @@ export class MockContext {
         this.realCreateElement = realCreateElement;
         this.ctx = {
             nodeCanvas,
+            getRenderContext2D: () => this.ctx.nodeCanvas.getContext('2d') as unknown as CanvasRenderingContext2D,
```
Note: The TS compiler is reporting that the real and mock canvases are not compatible (which isn't what I would expect from a full mock). This isn't an ideal solution, but it's better that adding `as` in every `.test.ts` file that uses the `CanvasRenderingContext2D` object.

olegat committed:
AG-NONE Fix compiler error with render context

olegat committed:
AG-NONE Fix error with defaultTimeTickFormat()

olegat committed:
AG-NONE Fix imports after rebase

olegat committed:
AG-NONE Fix imports

olegat committed:
AG-NONE Fix timeScale.test.ts

alantreadway commented on packages/ag-charts-community/src/options/series/cartesian/cartesianSeriesTooltipOptions.ts:
```diff
@@ -15,6 +15,7 @@ export interface AgCartesianSeriesTooltipRendererParams<TDatum = any> extends Ag
 export interface AgErrorBoundSeriesTooltipRendererParams {
     /** The key to use to retrieve lower bound error values from the x axis data. */
     xLowerKey?: string;
+    // TODO(olegat) shouldn't these values be removed??
```
Nit: As this is the public API, it might be better to avoid adding these types of internal note here in case they leak.

olegat replied:
Well spotted! Done.

alantreadway commented on packages/ag-charts-community/tsconfig.json:
```diff
@@ -11,7 +11,7 @@
   },
   "files": [],
-  "include": [],
+  "include": ["jest.config.ts", "jest.setup.ts", "src/**/*.ts"],
```
Can you check the impact of this on the output in `dist/types/` - I'm just concerned this could lead to test-typings being emitted there?

olegat replied:
I'm seeing `*.d.test.ts` files in the `dist/types/` directory. However I'm also seeing that on `latest`, so it's unclear whether these changes are actually impacting the output.

Shall we just merge this PR and clean up the `dist/types` directory in a follow-up?

alantreadway commented on packages/ag-charts-community/tsconfig.types.json:
```diff
@@ -5,6 +5,6 @@
         "types": ["node"]
     },
-    "include": ["src/*.ts"],
+    "include": ["jest.config.ts", "jest.setup.ts", "src/**/*.ts"],
```
Ditto

olegat committed:
AG-NONE Remove TODO comment

olegat committed:
AG-NONE Add a separate build target for test types

This avoids *.d.test.ts files from being generated in the dist/types/
output directory.

alantreadway committed:
Fix-up test build targets.

olegat committed:
nx format

alantreadway committed:
Link in test build targets.

alantreadway committed:
Test fixes.

alantreadway committed:
Temporarily disable ag-charts-enterprise:build:test

alantreadway committed:
Fix test lint errors

alantreadway committed:
Build target cleanup.

alantreadway committed:
Fix examples.

alantreadway committed:
Fix broken tests.
